scaffoldmaker/utils/vector.py

```diff
@@ -38,3 +39,34 @@ def setMagnitude(v, mag):
     '''
     scale = mag/math.sqrt(sum(c*c for c in v))
     return [ c*scale for c in v ]
+
+def getDiagonal(m):
```

Comment:
This is used as follows:
R = getDiagonal([cosa, cosa, cosa])
Does that mean it should be renamed createDiagonalMatrix?
Creates a square matrix with the supplied values on its diagonal and zeros everywhere else.
:param d: List of values on diagonal.
:return: Square matrix as list (columns) of lists (rows) of values.